app/src/main/java/com/example/sabonit/CartAdapter.java

```diff
@@ -0,0 +1,159 @@
+package com.example.sabonit;
+//******************imports*******************************/
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.Intent;
+import android.text.SpannableString;
+import android.text.method.ScrollingMovementMethod;
+import android.text.style.RelativeSizeSpan;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+//******************pizza adapter*******************************/
+
+
+/**
+ * A class which represents an adapter of Pizza objects, which extends
+ * the RecyclerView.Adapter<CartAdapter.ViewHolder> class.
+ */
+public class CartAdapter extends RecyclerView.Adapter<CartAdapter.ViewHolder> {
```

Comment:
in the cart activity screen, all the functionalty happens from here?

Reply:
Yes, all the functionality (of the CartActivity) happens by the CartAdapter class which extends the RecyclerView.Adapter<CartAdapter.ViewHolder> class